src/no-status-code-assert.ts

```diff
@@ -0,0 +1,104 @@
+// no-status-code-assert.ts
+
+/*
+ * Copyright (c) 2022-2024 Check Digit, LLC
+ *
+ * This code is licensed under the MIT license (see LICENSE.txt for details).
+ */
+
+import { AST_NODE_TYPES, ESLintUtils, TSESLint, TSESTree } from '@typescript-eslint/utils';
+import getDocumentationUrl from './get-documentation-url';
+
+export const ruleId = 'no-status-code-assert';
+const NO_STATUS_CODE_ASSERT = 'NO_STATUS_CODE_ASSERT';
+const keywords = ['status', 'code', 'StatusCodes', 'statusCode'];
+
+const createRule = ESLintUtils.RuleCreator((name) => getDocumentationUrl(name));
+
+/**
+ * Checks if a given AST node contains any identifier, member expression, or binary expression
+ * that includes the keywords 'status', 'code', 'StatusCodes', or 'statusCode'.
+ *
+ * @param arg - The AST node to check.
+ * @returns `true` if the node or its sub-nodes contain any of the specified keywords, otherwise `false`.
+ */
+const hasStatusCodeOrValue = (arg: TSESTree.Node): boolean => {
+  const checkName = (name?: string): boolean =>
+    name !== undefined && keywords.some((keyword) => name.toLowerCase().includes(keyword));
```

Comment:
the usage of `name.toLowerCase()` seems contradict with the keyword const values which contains upper case values.